src/components/preview/preview.component.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { MarginProps } from "styled-system";
+
+import PreviewPlaceholder, {
+  PreviewPlaceholderProps,
+} from "./__internal__/preview-placeholder.component";
+import { StyledPreview } from "./preview.style";
+import { filterStyledSystemMarginProps } from "../../style/utils";
+
+export interface PreviewProps
+  extends Partial<Omit<PreviewPlaceholderProps, "index">>,
+    MarginProps {
+  /** Children content to render in the component. */
+  children?: React.ReactNode;
+  /* Provides more control over when in a loading state. */
+  loading?: boolean;
+}
+
+export const Preview = ({
+  children,
+  loading,
+  lines = 1,
+  ...props
+}: PreviewProps) => {
+  const marginProps = filterStyledSystemMarginProps(props);
+  const hasPlaceholder = typeof loading !== "undefined" ? loading : !children;
+
+  if (hasPlaceholder) {
+    const placeholders = [];
+
+    for (let i = 1; i <= lines; i++) {
+      placeholders.push(
+        <PreviewPlaceholder key={i} index={i} lines={lines} {...props} />
+      );
+    }
```

Comment:
What do you think about creating a function out of this or declaring this declaratively so we can memorise the value to save rerenders?